
diff --git a/fs/smb/server/oplock.c b/fs/smb/server/oplock.c
index 1b8c3482d1e4..45af4651dcc7 100644
--- a/fs/smb/server/oplock.c
+++ b/fs/smb/server/oplock.c
@@ -765,10 +765,7 @@ static bool wait_for_break_ack(struct oplock_info *opinfo)

 static void wake_up_oplock_break(struct oplock_info *opinfo)
 {
-       clear_bit_unlock(0, &opinfo->pending_break);
-       /* memory barrier is needed for wake_up_bit() */
-       smp_mb__after_atomic();
-       wake_up_bit(&opinfo->pending_break, 0);
+       clear_and_wake_up_bit(0, &opinfo->pending_break);
 }

 static bool oplock_break_set_ack_wait(struct oplock_info *opinfo)
